File: code/example_code/load_data.py
"""
This script loads a dataset from a CSV.gz file and prints the first few rows.

Usage:
    python load_data.py <input_file>

Arguments:
    input_file: str
        The path to the input CSV.gz file.

Example:
    python load_data.py data/raw/sample.csv.gz
"""

import pandas as pd
import sys

input_file = sys.argv[1]

df = pd.read_csv(input_file)

print(df.head())

File: code/example_code/load_data.R
#!/usr/bin/env Rscript

# This script loads a dataset from a CSV.gz file and prints the first few rows.
#
# Usage:
#     Rscript load_data.R <input_file>
#
# Arguments:
#     input_file: character
#         The path to the input CSV.gz file.
#
# Example:
#     Rscript load_data.R data/raw/sample.csv.gz

# Load required package
if (!require(readr)) {
  install.packages("readr")
  library(readr)
}

# Get command line arguments
args <- commandArgs(trailingOnly = TRUE)

if (length(args) != 1) {
  stop("Please provide the input file path as an argument")
}

input_file <- args[1]

# Read the data
df <- read_csv(input_file)

# Print the first few rows
print(head(df))